include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,619 @@
+
+	/** Supported hash algorithms */
+	odp_comp_hash_algos_t hash_algos;
+
+	/** Synchronous compression mode support (ODP_COMP_OP_MODE_SYNC) */
+	odp_support_t sync;
+
+	/** Aynchronous compression mode support (ODP_COMP_OP_MODE_SSYNC) */
+	odp_support_t async;
+} odp_comp_capability_t;
+
+/**
+ * Hash algorithm capabilities
+ */
+typedef struct odp_comp_hash_alg_capability_t {
+	/** Digest length in bytes */
+	uint32_t digest_len;
+} odp_comp_hash_alg_capability_t;
+
+/**
+ * Compression dictionary.
+ *
+ * The dictionary should consist of strings (byte sequences) that are likely to
+ * be encountered later in the data to be compressed, with the most commonly
+ * used strings preferably put towards the end of the dictionary.
+ * Using a dictionary is most useful when the data to be compressed is short
+ * and can be predicted with good accuracy; the data can then be compressed
+ * better than with the default empty dictionary.
+ *
+ * The compression and decompression must use exactly the same dictionary.
+ * The optimal dictionaries are application specific
+ * The dictionary may be fixed or may be chosen among a certain number of
+ * predefined dictionaries, according to the kind of input data
+ *
+ * Refer RFC 1950
+ */
+typedef struct odp_comp_dict_t {
+	/** pointer to byte array */
+	uint8_t *buf;
+	/** length of the dictionary */
+	uint32_t len;
+} odp_comp_dict_t;
+
+/**
+ * Compression algorithm capabilities
+ */
+typedef struct odp_comp_alg_capability_t {
+
+	/** Maximum length of dictionary supported for this algorithm
+	 * Value of zero indicates dictionary is not supported.
+	 */
+	uint32_t dict_len;
+
+	/** Maximum compression level supported by implementation of this
+	 * algorithm. Indicates number of compression levels supported by
+	 * implementation. Valid range from (1 ... max_level)
+	 */
+	uint32_t max_level;
+
+	/** Supported hash algorithms */
+	odp_comp_hash_algos_t hash_algo;
+} odp_comp_alg_capability_t;
+
+/**
+ * Compression Huffman type. Used by DEFLATE algorithm
+ */
+typedef enum odp_comp_huffman_code {
+	/** Fixed Huffman code */
+	ODP_COMP_HUFFMAN_FIXED,
+
+	/** Dynamic Huffman code */
+	ODP_COMP_HUFFMAN_DYNAMIC,
+
+	/** Default huffman code selected by implementation */
+	ODP_COMP_HUFFMAN_DEFAULT,
+} odp_comp_huffman_code_t;
+
+/**
+ * Compression DEFLATEe algorithm parameters.
+ * Also initialized by other deflate based algorithms , ex. ZLIB
+ */
+typedef struct odp_comp_alg_def_param {
+	/**
+	 * Compression level
+	 *
+	 * Valid range is integer between (0 ... max_level)
+	 * level supported by the implementation.
+	 *
+	 * where,
+	 * 0 - implemention default
+	 *
+ 	 * 1 - fastest compression i.e. output produced at
+ 	 * best possible speed at the expense of compression quality
+	 *
+	 * max_level - High quality compression
+	 *
+	 * @see 'max_level' in odp_comp_alg_capability_t
+	 */
+	uint32_t comp_level;
+
+	/** huffman code to use */
+	odp_comp_huffman_code_t comp_code;
+} odp_comp_alg_def_param_t;
```

Comment:
Better type name is odp_comp_deflate_param_t. odp_comp_alg_def_param_t can be mixed to "algorithm default" parameters...